pages/account.js

```diff
@@ -0,0 +1,255 @@
+import React, {Component} from 'react';
+import NavbarBootstrap from '../components/NavbarBootstrap';
+import Container from 'react-bootstrap/Container';
+import Card from 'react-bootstrap/Card';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Nav from 'react-bootstrap/Nav';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import CarRow from '../components/CarRow';
+import axios from 'axios';
+
+const host = 'http://localhost';
+const port = 3001;
+
+class Account extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isAuthenticated: false,
+            name: '',
+            email: '',
+            newName: '',
+            newEmail: '',
+            succeededProfileUpdate: false,
+            failedProfileUpdate: false,
+            oldPasswordInput: '',
+            newPasswordInput: '',
+            succeededPasswordUpdate: false,
+            failedPasswordUpdate: false,
+        };
+    }
+
+    componentDidMount = () => {
+        const authToken = window.localStorage.getItem('auth_token');
+
+        if (!authToken) {
+            this.setState({ isAuthenticated: false });
+        } else {
+            axios.post(`${host}:${port}/authstatus`, { headers: { 'Authorization' : `Bearer ${authToken}` } }).then(res => {
+                if (res.status === 200) {
+                    const { name, email } = res.data.authData;
+                    this.setState({
+                        isAuthenticated: true,
+                        name,
+                        email,
+                        newName: name,
+                        newEmail: email,
+                    });
+                }
+            }).catch(err => {});
+        }
+    }
+
+    updateProfile = () => {
+        const { email, newName, newEmail } = this.state;
```

Comment:
we need to add a check here to make sure the newname OR the newemail (at least one of them) is different than the old one - otherwise no point in making a network request

Reply:
Oh! good catch. Totally forgot about that. Thank! 😊